src/daft-io/src/s3_like.rs

```diff
@@ -196,6 +196,36 @@ impl From<Error> for super::Error {
     }
 }
 
+/// Retrieves an S3Config from the environment by leveraging the AWS SDK's credentials chain
+pub(crate) async fn s3_config_from_env() -> super::Result<S3Config> {
+    let default_s3_config = S3Config::default();
+    let (anonymous, s3_conf) = build_s3_conf(&default_s3_config, None).await?;
+    let (key_id, access_key, session_token) = match s3_conf
+        .credentials_cache()
+        .provide_cached_credentials()
+        .await
+    {
+        Ok(creds) => (
+            Some(creds.access_key_id().to_string()),
+            Some(creds.secret_access_key().to_string()),
+            creds.session_token().map(|t| t.to_string()),
+        ),
+        Err(_) => (None, None, None),
```

Comment:
shouldn't be log this error or populate it up if the user is asking for an `S3Config` and we're unable to deliver it

Reply:
Yes you're right, I think the right behavior is to populate it upwards.